Tavisca.Bootcamp.LanguageBasics.Exercise1/Program.cs

```diff
@@ -14,16 +14,83 @@ static void Main(string[] args)
             Console.ReadKey(true);
         }
 
-        private static void Test(string args, int expected)
-        {
+        private static void Test(string args, int expected)
+        {
             var result = FindDigit(args).Equals(expected) ? "PASS" : "FAIL";
-            Console.WriteLine($"{args} : {result}");
+            Console.WriteLine($"{args} : {result}");
         }
 
-        public static int FindDigit(string equation)
-        {
+        public static int FindDigit(string equation)
+        {
             // Add your code here.
-            throw new NotImplementedException();
+
+            //Spliting the Equation.
+            String[] terms=equation.Split(new char[]{'*','='});
+
+            //Checking for the term with missing digit.
+            int pos=-1;
+            if(terms[0].Contains("?"))
+                pos=0;
+            else if(terms[1].Contains("?"))
+                pos=1;
+            else
+                pos=2;
+            int a=0,b=0,ccc=0;
+
+            //Finding the exact missing digit.
+            if(pos==0)
+            {
+                a=int.Parse(terms[1]);
+                b=int.Parse(terms[2]);
+                double cc=(double)b/a;
+                String c=cc.ToString();
+                //(cc%1==0) to check if the term is decimal; Used mainly for Test 5 type of cases
+                if(c.Length==terms[0].Length&&(cc%1==0))
+                {
+                    return int.Parse(c[terms[0].IndexOf('?')].ToString());
+                }
+            }
+            else if(pos==1)
+            {
+                a=int.Parse(terms[0]);
+                b=int.Parse(terms[2]);
+                double cc=(double)b/a;
+                String c=cc.ToString();
+                //(cc%1==0) to check if the term is decimal; Used mainly for Test 5 type of cases
+                if(c.Length==terms[1].Length&&(cc%1==0))
+                {
+                    return int.Parse(c[terms[1].IndexOf('?')].ToString());
+                }
+            }
+            else if(pos==2)
+            {
+                a=int.Parse(terms[0]);
+                b=int.Parse(terms[1]);
+                ccc=a*b;
+                String c=ccc.ToString();
+                if(c.Length==terms[2].Length)
+                {
+                    return int.Parse(c[terms[2].IndexOf('?')].ToString());
+                }
+            }
+
+            //If none of the above.
+            return -1;
+
+            throw new NotImplementedException();

```

Comment:
What is the use of this? You have implemented this function now you can remove this method call.